### PR TITLE
new: add new types for `int` and `float`

### PR DESCRIPTION
**Pull request by KennedyTedesco**

### CODE REVIEW

File: src/tree/definition/type.rs [hunk 1]
```diff
@@ -19,6 +19,34 @@ pub struct TypeAliasDefinition {
     pub semicolon: usize,
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Hash, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "snake_case", tag = "type", content = "value")]
+pub enum SignedIntegerType {
```
Comment (azjezz) [inline src/tree/definition/type.rs]:
```suggestion
pub enum SignedIntegerTypeDefinition {
```

**Comment by azjezz** [comment]
the int/float type structures need to implement node

**Comment by KennedyTedesco** [comment]
Done! :)

File: src/tree/definition/type.rs [hunk 2]
```diff
@@ -112,6 +142,108 @@ impl Node for TypeAliasDefinition {
...
+    fn children(&self) -> Vec<&dyn Node> {
+        match self {
+            FloatingPointType::Default(keyword)
+            | FloatingPointType::F64(keyword)
+            | FloatingPointType::F32(keyword) => vec![keyword],
+        }
+    }
+
+    fn get_description(&self) -> String {
+        "floating point type".to_string()
```
Comment (azjezz) [inline src/tree/definition/type.rs]:
```suggestion
        "floating point type definition".to_string()
```

File: src/tree/definition/type.rs [hunk 3]
```diff
@@ -135,6 +265,25 @@ impl Node for TypeDefinition {
...
             | TypeDefinition::Interface(keyword, _)
             | TypeDefinition::Iterable(keyword, _) => keyword.initial_position(),
+            TypeDefinition::SignedInteger(signed) => match signed {
+                SignedIntegerType::Default(keyword)
+                | SignedIntegerType::I128(keyword)
+                | SignedIntegerType::I64(keyword)
+                | SignedIntegerType::I32(keyword)
+                | SignedIntegerType::I16(keyword)
+                | SignedIntegerType::I8(keyword) => keyword.initial_position(),
+            },
```
Comment (azjezz) [inline src/tree/definition/type.rs]:
```suggestion
            TypeDefinition::SignedInteger(signed) => signed.initial_position(),
```

**Review by azjezz — CHANGES_REQUESTED** [review]

**Comment by azjezz** [comment]
awesome! thank you 🙏

**Pull request merged**